pyk/src/pyk/k2lean4/model.py

```diff
@@ -7,6 +7,9 @@
 
 if TYPE_CHECKING:
     from collections.abc import Iterable
+    from typing import Final
+
+_LEAN_KEYWORDS: Final = {'ite', 'end', 'where'}  # Words that cannot be a the name of a declaration
```

Comment:
`end` and `where` are keywords, `ite` is probably just a predefined function.

Reply:
I changed it to `_LEAN_WORDS`. Let me know if this notation is not desired, or if we should separate into two sets depending on keyword or definition.